flag-app/flag-app-project/src/components/Filter.tsx

```diff
@@ -1,24 +1,35 @@
-import { useDarkMode } from "../hooks";
+import { useCountries, useDarkMode } from "../hooks";
 import { RiArrowDropDownLine } from "react-icons/ri";
 
 export const Filter = () => {
   const { mode } = useDarkMode();
+  const { filterValue, handleFilter } = useCountries();
+
+  const baseClassName =
+    "pl-5 pt-5 pb-5 bg-white w-56 rounded-lg shadow-sm appearance-none";
+  const darkModeClassName =
+    mode === "dark" ? "bg-darkModeBlack shadow-white" : "bg-white";
+
+  const regions = [
+    { value: "", label: "Filter by Region" },
+    { value: "africa", label: "Africa" },
+    { value: "america", label: "America" },
+    { value: "europe", label: "Europe" },
+    { value: "oceania", label: "Oceania" },
+  ];
+
   return (
     <div className="relative">
       <select
-        // value={value}
-        // onChange={onChange}
-        className={`pl-5 pt-5 pb-5 bg-white w-56 rounded-lg shadow-sm appearance-none ${
-          mode === "dark" ? "bg-darkModeBlack shadow-white" : "bg-white"
-        }`}
+        value={filterValue}
+        onChange={(e) => handleFilter(e.target.value)}
```

Comment:
I would just move this function to a separate place, because if you had input validation, this would become a multiline function, making this hard to read